src/app/(sidebar)/xdr/view/page.tsx

```diff
@@ -54,6 +54,23 @@ export default function ViewXdr() {
     }
   }, [isLatestTxnSuccess, latestTxn, updateXdrBlob, updateXdrType]);
 
+  useEffect(() => {
+    if (isXdrInit && xdr.blob) {
+      try {
+        const guessed = StellarXdr.guess(xdr.blob);
+
+        // enable xdr type selector to use the guessed type as default
+        if (guessed.includes("TransactionEnvelope")) {
```

Comment:
Added a function to select the default guessed value for the xdr type dropdown. I found it useful when I was testing `ScVal` xdr

Reply:
Previously, we decided against pre-selecting the guessed type [here](https://github.com/stellar/laboratory/issues/949#issuecomment-2338908193). Let's remove it. :pray: